tests/misc/project_tree.py

```diff
@@ -0,0 +1,28 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) Pootle contributors.
+#
+# This file is a part of the Pootle project. It is distributed under the GPL3
+# or later license. See the LICENSE file for a copy of the license and the
+# AUTHORS file for copyright and authorship information.
+
+import pytest
+
+from pootle_app.project_tree import direct_language_match_filename
+
+
+@pytest.mark.parametrize('language_code, path_name, matched', [
+    (u'pt_BR', u'/path/to/file-pt_BR.po', True),
+    (u'br', u'/path/to/file-pt_BR.po', False),
+    (u'pt_BR', u'/path/to/pt_BR.po', True),
+    (u'br', u'/path/to/br.po', True),
+    (u'br', u'/path/to/pt-br.po', True),
```

Comment:
If case is relevant perhaps it is also a good idea to test for u'/path/to/file-pt-BR.po'